### ISSUE TITLE
Check for expired lock on first iteration rather than waiting for LEASE_DURATION milliseconds

### ISSUE THREAD

**Issue by jazir1979** [awslabs/dynamodb-lock-client#8]
Apologies if I'm not understanding this properly, but the behaviour I'm seeing is this:

1) A lock is put in place and never gets released but it does expire after the lease period ends
2) If another client attempts to acquire this lock, the acquireLock(...) method on its first loop iteration always sleeps for the lease duration, rather than doing an upfront check to determine that the lock had already expired

Is this intentional? In my case it's failing to work with another library (spring-integration-aws) which expects a result from the acquireLock() method within 10 seconds (when my lease duration is set to 20 seconds).

**Comment by artembilan** [comment]
I also find this logic as a bit odd: we don't check existing lock status on the first iteration and we may not pass this condition:
```
if (LockClientUtils.INSTANCE.millisecondTime() - currentTimeMillis > millisecondsToWait) {
          throw new LockNotGrantedException("Didn't acquire lock after sleeping for " + (LockClientUtils.INSTANCE.millisecondTime() - currentTimeMillis) + " milliseconds");
}
```
To go to the ` Thread.sleep(refreshPeriodInMilliseconds);` and to the second iteration. 
The `millisecondsToWait` might be small enough and DynamoDB request might be long enough do not let us to overcome that condition.
Therefore it would be great to check an expiration status on existing lock even from the first iteration. We definitely might even not need extra iteration and that unnecessary sleep in between.

Also the current logic doesn't meet `boolean tryLock(long time, TimeUnit unit) throws InterruptedException;` contract. 
I may agree about some timeout approximation during request to the DB and back, but there is definitely no any reason for me to wait for the lock more than `tryLock()` requests from me.

Originally I did a mistake like:
```
long timeToWaitForLock = System.currentTimeMillis() - start + TimeUnit.MILLISECONDS.convert(time, unit) - DynamoDbLockRegistry.this.leaseDuration;

this.acquireLockOptionsBuilder
			.withAdditionalTimeToWaitForLock(timeToWaitForLock)
			.withRefreshPeriod(0);
```
I'm really not interested to wait for the `leaseDuration` (remember: `tryLock()` contract ?), so i decided to minus it, but I missed the fact that you don't check status on the first iteration, therefore I failed to meet a condition for the second iteration.

Anyway, independently of my misinterpretation, there is that flaw, when we always fail with the condition because a `millisecondsToWait` is so small in comparison with the request time, that we never know an existing lock state.

Thanks
